chirp/drivers/uvk5.py

```diff
@@ -0,0 +1,1339 @@
+            firmware = "To get the firmware version please download"
+            "the image from the radio first"
+        else:
+            firmware = self.FIRMWARE_VERSION
+
+        val = RadioSettingValueString(0, 128, firmware)
+        val.set_mutable(False)
+        rs = RadioSetting("fw_ver", "Firmware Version", val)
+        roinfo.append(rs)
+
+        # Driver version
+        val = RadioSettingValueString(0, 128, DRIVER_VERSION)
+        val.set_mutable(False)
+        rs = RadioSetting("driver_ver", "Driver version", val)
+        roinfo.append(rs)
+
+        return top
+
+    # Store details about a high-level memory to the memory map
+    # This is called when a user edits a memory in the UI
+    def set_memory(self, mem):
+        number = mem.number-1
+
+        # Get a low-level memory object mapped to the image
+        _mem = self._memobj.channel[number]
+        _mem4 = self._memobj
+        # empty memory
+        if mem.empty:
+            _mem.set_raw("\xFF" * 16)
+            if number < 200:
+                _mem2 = self._memobj.channelname[number]
+                _mem2.set_raw("\xFF" * 16)
+                _mem4.channel_attributes[number] = 0x0f
+            return mem
+
+        # clean the channel memory, restore some bits if it was used before
+        if _mem.get_raw()[0] == "\xff":
+            # this was an empty memory
+            _mem.set_raw("\x00" * 16)
+        else:
+            # this memory was't empty, save some bits that we don't know the
+            # meaning of, or that we don't support yet
+            prev_0a = ord(_mem.get_raw()[0x0a]) & SAVE_MASK_0A
+            prev_0b = ord(_mem.get_raw()[0x0b]) & SAVE_MASK_0B
+            prev_0c = ord(_mem.get_raw()[0x0c]) & SAVE_MASK_0C
+            prev_0d = ord(_mem.get_raw()[0x0d]) & SAVE_MASK_0D
+            prev_0e = ord(_mem.get_raw()[0x0e]) & SAVE_MASK_0E
+            prev_0f = ord(_mem.get_raw()[0x0f]) & SAVE_MASK_0F
+            _mem.set_raw("\x00" * 10 +
+                         chr(prev_0a) + chr(prev_0b) + chr(prev_0c) +
+                         chr(prev_0d) + chr(prev_0e) + chr(prev_0f))
+
+        if number < 200:
+            _mem4.channel_attributes[number] = 0x0f
+
+        # find band
+        band = _find_band(mem.freq)
+        if band is False:
+            #    raise errors.RadioError(
+            #            "Frequency is outside the supported bands")
+            return mem
+
+        # mode
+        if mem.mode == "AM":
+            _mem.flags1 = _mem.flags1 | FLAGS1_ISAM
+            _mem.flags2 = _mem.flags2 & ~FLAGS2_BANDWIDTH
+        else:
+            _mem.flags1 = _mem.flags1 & ~FLAGS1_ISAM
+            if mem.mode == "NFM":
+                _mem.flags2 = _mem.flags2 | FLAGS2_BANDWIDTH
+            else:
+                _mem.flags2 = _mem.flags2 & ~FLAGS2_BANDWIDTH
+
+        # frequency/offset
+        _mem.freq = mem.freq/10
+        _mem.offset = mem.offset/10
+
+        if mem.duplex == "off" or mem.duplex == "":
```

Comment:
You're not handling `duplex=off` here. Either don't expose that you support it in `valid_duplexes` (which you're not) do and handle it properly. A duplex of `off` means "Inhibit TX" in CHIRP. Treating it the same as no duplex (i.e. simplex) is wrong.